File: java-samples/Advanced/snakesandladders-cordapp/workflows/src/main/java/net/corda/samples/snl/flows/QueryOracleForDiceRollFlow.java
package net.corda.samples.snl.flows;

import co.paralleluniverse.fibers.Suspendable;
import net.corda.core.flows.FlowException;
import net.corda.core.flows.FlowLogic;
import net.corda.core.flows.StartableByRPC;
import net.corda.core.identity.CordaX500Name;
import net.corda.core.identity.Party;
import net.corda.samples.snl.oracle.flows.DiceRollerFlow;

@StartableByRPC
public class QueryOracleForDiceRollFlow extends FlowLogic<Integer> {

    private String player;

    public QueryOracleForDiceRollFlow(String player) {
        this.player = player;
    }

    @Override
    @Suspendable
    public Integer call() throws FlowException {
        Party oracle = getServiceHub().getNetworkMapCache()
                .getNodeByLegalName(CordaX500Name.parse("O=Oracle,L=Mumbai,C=IN")).getLegalIdentities().get(0);

        return subFlow(new DiceRollerFlow(player, oracle));
    }
}

File: kotlin-samples/Advanced/snakesandladders-cordapp/workflows/src/main/kotlin/net/corda/samples/snl/flows/QueryOracleForDiceRollFlow.kt
package net.corda.samples.snl.flows

import co.paralleluniverse.fibers.Suspendable
import net.corda.core.flows.FlowException
import net.corda.core.flows.FlowLogic
import net.corda.core.flows.StartableByRPC
import net.corda.core.identity.CordaX500Name
import net.corda.samples.snl.oracle.flows.DiceRollerFlow

@StartableByRPC
class QueryOracleForDiceRollFlow(private val player: String) : FlowLogic<Int>() {
    @Suspendable
    @Throws(FlowException::class)
    override fun call(): Int {
        val oracle = serviceHub.networkMapCache
                .getNodeByLegalName(CordaX500Name.parse("O=Oracle,L=Mumbai,C=IN"))!!.legalIdentities[0]
        return subFlow(DiceRollerFlow(player, oracle))!!
    }
}